[CI] RepositoryAnalysisSuccessIT testRepositoryAnalysis failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/5q2ukbpqbhz7i/tests/:x-pack:plugin:snapshot-repo-test-kit:internalClusterTest/org.elasticsearch.repositories.blobstore.testkit.RepositoryAnalysisSuccessIT/testRepositoryAnalysis

**Reproduction line:**
```
./gradlew ':x-pack:plugin:snapshot-repo-test-kit:internalClusterTest' --tests "org.elasticsearch.repositories.blobstore.testkit.RepositoryAnalysisSuccessIT.testRepositoryAnalysis" -Dtests.seed=BE53761FA83A5950 -Dtests.locale=ca -Dtests.timezone=America/Port-au-Prince -Druntime.java=20
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.repositories.blobstore.testkit.RepositoryAnalysisSuccessIT&tests.test=testRepositoryAnalysis

**Failure excerpt:**
```
org.elasticsearch.repositories.RepositoryVerificationException: [test-repo] analysis failed, you may need to manually remove [temp-analysis-0Fi4UgK2T72QJxKdRInE2Q]

  at __randomizedtesting.SeedInfo.seed([BE53761FA83A5950:476ED77176A16AB9]:0)

  Caused by: org.elasticsearch.repositories.RepositoryVerificationException: [test-repo] register [test-register-EZnfK-0gTm-NSGWxRGkMzw] should have value [3] but instead had value [OptionalLong.empty]


```

--- a/x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalysisSuccessIT.java
+++ b/x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/RepositoryAnalysisSuccessIT.java
@@ -424,6 +424,7 @@ public class RepositoryAnalysisSuccessIT extends AbstractSnapshotIntegTestCase {
 
         @Override
         public void compareAndExchangeRegister(String key, long expected, long updated, ActionListener<OptionalLong> listener) {
+            firstRegisterRead.set(false);
             if (updated != -1 && randomBoolean() && randomBoolean()) {
                 // updated != -1 so we don't fail the final check because we know there can be no concurrent operations at that point
                 listener.onResponse(OptionalLong.empty());

